File: exercise_1/task_5/minmax.cpp
#include <iostream>
using namespace std;
void minmax(int data[], int length, int *min, int *max);

int main()
{
  int data[] = {4, 8, 23, -17, 2, -6};
  int length = sizeof(data) / sizeof(data[0]);
  int min = 0, max = 0;
  minmax(data, length, &min, &max);

  cout << "Min=" << min << " Max=" << max;
  return 0;
}

void minmax(int data[], int length, int *min, int *max)
{
  *min = data[0];
  *max = data[0];

  for (int i = 1; i < length - 1; i++)
  {
    if (data[i] < *min)
      *min = data[i];
    if (data[i] > *max)
      *max = data[i];
  }
}

File: exercise_1/task_4/MinMax.java
package exercise_1.task_4;

// ZF: weil java die primitiven daten typen by value und nicht by reference übergibt 

class MinMax {
  public static void main(String[] args) {
    int[] data = { 4, 8, 23, -17, 2, -6 };
    System.out.println("Minimum: " + min(data) + ", Maximum: " + max(data));
  }

  static int min(int[] x){
    int min = x[0];
    for (int i : x) if (i < min) min = i;
    return min;
  }

  static int max(int[] x){
    int max = x[0];
    for (int i : x) if (i > max) max = i;
    return max;
  }
}